internal/pkg/build/sources/conveyorPacker_local_test.go

```diff
@@ -0,0 +1,128 @@
+// Copyright (c) 2018-2024, Sylabs Inc. All rights reserved.
+// This software is licensed under a 3-clause BSD license. Please consult the
+// LICENSE.md file distributed with the sources of this project regarding your
+// rights to use or distribute this software.
+
+package sources_test
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/sylabs/singularity/v4/internal/pkg/build/sources"
+	"github.com/sylabs/singularity/v4/internal/pkg/test/tool/require"
+	"github.com/sylabs/singularity/v4/internal/pkg/util/fs"
+	"github.com/sylabs/singularity/v4/internal/pkg/util/fs/squashfs"
+	"github.com/sylabs/singularity/v4/pkg/build/types"
+)
+
+func TestLocalPackerSquashfs(t *testing.T) {
+	require.Command(t, "mksquashfs")
+
+	tempDirPath, err := os.MkdirTemp("", "test-localpacker-squashfs")
+	if err != nil {
+		t.Fatalf("while creating temp dir: %v", err)
+	}
+	defer os.RemoveAll(tempDirPath)
+
+	// Image root directory
+	rootfs := filepath.Join(tempDirPath, "issue_3084_rootfs")
+
+	// Create directories
+	if err := os.Mkdir(rootfs, 0o755); err != nil {
+		t.Fatalf("while creating directory: %v", err)
+	}
+	if err := os.Mkdir(filepath.Join(rootfs, "tmp"), 0o755); err != nil {
+		t.Fatalf("while creating directory: %v", err)
+	}
+	if err := os.Mkdir(filepath.Join(rootfs, "var"), 0o755); err != nil {
+		t.Fatalf("while creating directory: %v", err)
+	}
+
+	// Create symlinks: /var/tmp -> /tmp , /var/log -> /tmp
+	if err := os.Symlink(filepath.Join(rootfs, "tmp"), filepath.Join(rootfs, "var", "tmp")); err != nil {
+		t.Fatalf("while creating symlink: %v", err)
+	}
+	if err := os.Symlink(filepath.Join(rootfs, "tmp"), filepath.Join(rootfs, "var", "log")); err != nil {
```

Comment:
`rootfs` is an absolute path to somewhere on the temporary folder, isn't it? I'd suggest to use the original one with `../tmp` to match the original issue or `/tmp` directly to match the comment.
```suggestion
	if err := os.Symlink(filepath.Join("..", "tmp"), filepath.Join(rootfs, "var", "tmp")); err != nil {
		t.Fatalf("while creating symlink: %v", err)
	}
	if err := os.Symlink(filepath.Join("..", "tmp"), filepath.Join(rootfs, "var", "log")); err != nil {
```

Reply:
Hi, thank you very much for the review and pointing out this issue. I'll check it soon.